exercises/07-LifecycleMethods.js.jsx

```diff
@@ -30,18 +30,41 @@ import React from 'react';
 // There are more lifecycle methods available.
 // Those three presented are commonly used.
 //
+
+// This could have done more than just console.log things
+
+
 // Extra task: Learn about componentWillUpdate. What's the difference between
 //             this and the componentDidUpdate method? Think about the possible
 //             use cases of this lifecycle method.
+// componentWillUpdate is called once it is known that a component should update. This shoud be used to prep for updated data
+// it SHOULD NOT trigger an update itself.
+// componentDidMount is called after the render method and the DOM operations can happen here.
+
+
 // Extra task: Learn about componentWillMount. How can it be useful?
 //             (Hint: Think about server-side rendering of React components)
+// componentWillMount is called before the render method and setting state here won't call a re-render. 
+// this can be used to initialize data.
+
+
 // Extra task: Learn about componentWillReceiveProps. How it can be used?
 //             Is it fired when you render a component for the first time?
+// This can be used to update the state depdning on the existing and upcoming props. It does not trigger another rendering
+// No, it is not fired on initial rendering.
+
+
+
 // Extra task: There is a method which directly modifies behavior of React
 //             itself - it's called shouldComponentUpdate.
 //             How can you use it to optimise rendering cycle of your
 //             React components? Learn about PureRenderMixin.
-//
+// shouldComponentUpdate and allows the coder to define if re--rendering is needed or can be skipped
+// this can be used to conditionally re-render components based on state change. So if part of the state changes
+// that does not affect a particular component, there is no need to re-render it.
```

Comment:
The new props are also passed in to this lifecycle method, which can also be analyzed to determine if a re-render should happen or not.